cffi_src/factory.go

```diff
@@ -135,10 +136,54 @@ func BuildRequest(input RequestInput) (*http.Request, *TLSClientError) {
 }
 
 // BuildResponse constructs a client response from a given HTTP response. The client response can then be sent to the interface consumer.
-func BuildResponse(sessionId string, withSession bool, resp *http.Response, cookies []*http.Cookie, isByteResponse bool) (Response, *TLSClientError) {
+func BuildResponse(sessionId string, withSession bool, resp *http.Response, cookies []*http.Cookie, input RequestInput) (Response, *TLSClientError) {
 	defer resp.Body.Close()
+	isByteResponse := input.IsByteResponse
+	var respBodyBytes []byte
+	var err error
+
+	if input.StreamOutputPath != nil {
```

Comment:
i would like to see the whole handling of writing the response to a file in a separate method.
would increase readability imho.

Reply:
Done! What do you think for the new version?